### PR TITLE
Update framework section on performance

### PR DESCRIPTION
**Pull request by OwenArnold**
Update to release notes based on recent changes #18906. I've compared both PerformanceTests and SystemTests in order to make these notes.

We may need to provide some additional/amended information again just prior to the release, but I think this is a good time to get something in at a time it's easy to measure the changes. And before we forget.

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
@OwenArnold Did you gather these data mainly from performance tests (apart from those where you mention system tests)? If so, can we double check data by hand for some real files? Not all performance tests have been written to equal standards, so before making public statements getting confirmation might be a good idea.

File: docs/source/release/v3.10.0/framework.rst [hunk 1]
```diff
@@ -48,6 +48,10 @@ MD Algorithms (VATES CLI)
 Performance
 -----------
 
+Following a large effort to change some of the core geometric aspects of Mantid, we are pleased to announce very significant improvements in performance as part of the 3.10.0 release. :ref:`ConvertUnits <algm-ConvertUnits>` for example is running >2x times faster than possible in our last major 3.9.0 release. Amongst others, :ref:`NormaliseVanadium <algm-NormaliseVanadium`, :ref:`MaskDetectorsInShape <algm-MaskDetectorsInShape>`,  :ref:`FindSXPeaks <algm-FindSXPeaks>`, all now give a 2x speedup. :ref:`ConvertSpectrumAxis <algm-ConvertSpectrumAxis>` and :ref:`AnvredCorrection <algm-AnvredCorrection` are examples of a few monitored algorithms that are demonstrating a more modest 10% improvement in speed over the last major mantid release. Aside from granular improvements at the algorithm level, we have observed that many of the Direct Inelastic technique area workflows are 2x faster than before.
```
Comment (SimonHeybrock) [inline docs/source/release/v3.10.0/framework.rst]: Maybe talk about "instrument" rather than "geometry"?

**Comment by OwenArnold** [comment]
@SimonHeybrock I've been careful about checking the system tests ensuring that we are cross comparing the same tests on the same OS, on the same machine. My findings support the results indicated by @martyngigg. Although I'm not reporting exact numbers, a few of the DirectInelastic workflows are reporting a 2x speedup.

For the performance tests, I have checked the tests themselves and the code (i.e. ballpark how much I would expect them to improve based on their reliance on positions/l1/l2 etc). Again I'm satisfied that what we are reporting is real.

**Comment by SimonHeybrock** [comment]
:shipit:

**Comment by SimonHeybrock** [comment]
@OwenArnold I remember a while ago you mentioned beam-center finding in SANS that might be affected by therse changes. Have you checked the performance there, or talked to Anton?

**Comment by OwenArnold** [comment]
That's a good point. I'll talk to @AntonPiccardoSelg about that

**Pull request merged**